packages/frontend/src/components/common/PivotTable/ValueCellMenu.tsx

```diff
@@ -1,22 +1,90 @@
-import { PivotValue } from '@lightdash/common';
+import { Field, PivotValue, TableCalculation } from '@lightdash/common';
 import { Menu, MenuProps } from '@mantine/core';
-import { IconCopy } from '@tabler/icons-react';
-import { FC } from 'react';
+import { IconArrowBarToDown, IconCopy, IconStack } from '@tabler/icons-react';
+import { FC, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { useApp } from '../../../providers/AppProvider';
+import { useTracking } from '../../../providers/TrackingProvider';
+import { EventName } from '../../../types/Events';
+import {
+    UnderlyingValueMap,
+    useMetricQueryDataContext,
+} from '../../MetricQueryData/MetricQueryDataProvider';
 import MantineIcon from '../MantineIcon';
 
 type ValueCellMenuProps = {
+    rowIndex: number;
+    colIndex: number;
+    item: Field | TableCalculation | undefined;
     value: PivotValue | null;
+    row: (PivotValue | null)[];
     onCopy: () => void;
+    getUnderlyingFieldValues: (
+        colIndex: number,
+        rowIndex: number,
+    ) => UnderlyingValueMap;
 } & Pick<MenuProps, 'opened' | 'onOpen' | 'onClose'>;
 
 const ValueCellMenu: FC<ValueCellMenuProps> = ({
     children,
-    // value,
+    rowIndex,
+    getUnderlyingFieldValues,
+    colIndex,
+    item,
+    value: pivotValue,
+    // row: pivotRow,
     opened,
     onOpen,
     onClose,
     onCopy,
 }) => {
+    const { track } = useTracking();
+    const { openUnderlyingDataModal, openDrillDownModel } =
+        useMetricQueryDataContext();
+    const { user } = useApp();
+    // TODO: get rid of this from here
+    const { projectUuid } = useParams<{ projectUuid: string }>();
```

Comment:
will do a proper cleanup in a follow-up PR
components like this should not read from the query string